python/lsst/pipe/tasks/subtractBrightStars.py

```diff
@@ -0,0 +1,359 @@
+            "visit",
+            "detector",
+        ),
+    )
+    outputBackgroundExposure = cT.Output(
+        doc="Exposure containing only the modelled bright stars.",
+        name="{outputBackgroundName}_calexp_background",
+        storageClass="ExposureF",
+        dimensions=(
+            "visit",
+            "detector",
+        ),
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+        if not config.doApplySkyCorr:
+            self.inputs.remove("skyCorr")
+
+
+class SubtractBrightStarsConfig(PipelineTaskConfig, pipelineConnections=SubtractBrightStarsConnections):
+    """Configuration parameters for SubtractBrightStarsTask"""
+
+    doWriteSubtractor = Field(
+        dtype=bool,
+        doc="Should an exposure containing all bright star models be written to disk?",
+        default=True,
+    )
+    doWriteSubtractedExposure = Field(
+        dtype=bool,
+        doc="Should an exposure with bright stars subtracted be written to disk?",
+        default=True,
+    )
+    magLimit = Field(
+        dtype=float,
+        doc="Magnitude limit, in Gaia G; all stars brighter than this value will be subtracted",
+        default=18,
+    )
+    warpingKernelName = ChoiceField(
+        dtype=str,
+        doc="Warping kernel",
+        default="lanczos5",
+        allowed={
+            "bilinear": "bilinear interpolation",
+            "lanczos3": "Lanczos kernel of order 3",
+            "lanczos4": "Lanczos kernel of order 4",
+            "lanczos5": "Lanczos kernel of order 5",
+            "lanczos6": "Lanczos kernel of order 6",
+            "lanczos7": "Lanczos kernel of order 7",
+        },
+    )
+    scalingType = ChoiceField(
+        dtype=str,
+        doc="How the model should be scaled to each bright star; implemented options are "
+        "`annularFlux` to reuse the annular flux of each stamp, or `leastSquare` to perform "
+        "least square fitting on each pixel with no bad mask plane set.",
+        default="leastSquare",
+        allowed={
+            "annularFlux": "reuse BrightStarStamp annular flux measurement",
+            "leastSquare": "find least square scaling factor",
+        },
+    )
+    badMaskPlanes = ListField(
+        dtype=str,
+        doc="Mask planes that, if set, lead to associated pixels not being included in the computation of "
+        "the scaling factor (`BAD` should always be included). Ignored if scalingType is `annularFlux`, "
+        "as the stamps are expected to already be normalized.",
+        # Note that `BAD` should always be included, as secondary detected
+        # sources (i.e., detected sources other than the primary source of
+        # interest) also get set to `BAD`.
+        default=("BAD", "CR", "CROSSTALK", "EDGE", "NO_DATA", "SAT", "SUSPECT", "UNMASKEDNAN"),
+    )
+    doApplySkyCorr = Field(
+        dtype=bool, doc="Apply full focal plane sky correction before extracting stars?", default=True
```

Comment:
Is this a black thing? I'd like it personally if these were in their own individual lines. One way to make black wrap it would be to give a trailing comma, as is the case in previous Fields.